backend/controller/evaljob_controller.go

```diff
@@ -0,0 +1,568 @@
+	job.Status.LastScheduleTime = &currentTime
+	if err := r.Status().Update(ctx, job); err != nil {
+		log.Error(err, "unable to update EvalJob status (pod creation done)")
+		return ctrl.Result{}, err
+	}
+	r.Recorder.Event(job, "Normal", "PodCreation",
+		fmt.Sprintf("the EvalJob %s in namespace %s created a pod",
+			job.Name,
+			job.Namespace))
+	log.Info("Successfully create a Pod for the Job")
+	// Check the pod after 10 seconds
+	return ctrl.Result{Requeue: true, RequeueAfter: time.Second * 10}, nil
+}
+
+func (r *EvalJobReconciler) checkScheduledPod(ctx context.Context, log logr.Logger, job *lmevalservicev1beta1.EvalJob) (ctrl.Result, error) {
+	pod, err := r.getPod(ctx, job)
+	if err != nil {
+		// a weird state, someone delete the corresponding pod? mark this as CompleteJobState
+		// with error message
+		job.Status.State = lmevalservicev1beta1.CompleteJobState
+		job.Status.Reason = lmevalservicev1beta1.FailedReason
+		job.Status.Message = err.Error()
+		if err := r.Status().Update(ctx, job); err != nil {
+			log.Error(err, "unable to update EvalJob status", "state", job.Status.State)
+			return ctrl.Result{}, err
+		}
+		r.Recorder.Event(job, "Warning", "PodMising",
+			fmt.Sprintf("the pod for the EvalJob %s in namespace %s is gone",
+				job.Name,
+				job.Namespace))
+		log.Error(err, "since the job's pod is gone, mark the job as complete with error result.")
+		return ctrl.Result{}, err
+	}
+
+	if pod.Status.ContainerStatuses == nil {
+		// wait for the pod to initialize and run the containers
+		return ctrl.Result{Requeue: true, RequeueAfter: time.Second * 10}, nil
+	}
+
+	for _, cstatus := range pod.Status.ContainerStatuses {
+		if cstatus.Name == "main" {
+			if cstatus.LastTerminationState.Terminated == nil {
+				return ctrl.Result{Requeue: true, RequeueAfter: time.Second * 10}, nil
+			} else {
+				if cstatus.LastTerminationState.Terminated.ExitCode == 0 {
+					job.Status.State = lmevalservicev1beta1.CompleteJobState
+					job.Status.Reason = lmevalservicev1beta1.SucceedReason
+				} else {
+					job.Status.State = lmevalservicev1beta1.CompleteJobState
+					job.Status.Reason = lmevalservicev1beta1.FailedReason
+					job.Status.Message = cstatus.LastTerminationState.Terminated.Reason
+
+				}
+				if err := r.Status().Update(ctx, job); err != nil {
+					log.Error(err, "unable to update EvalJob status", "state", job.Status.State)
+					return ctrl.Result{}, err
+				}
+				r.Recorder.Event(job, "Normal", "PodCompleted",
+					fmt.Sprintf("The pod for the EvalJob %s in namespace %s has completed",
+						job.Name,
+						job.Namespace))
+				return ctrl.Result{}, nil
+			}
+		}
+	}
+	return ctrl.Result{Requeue: true, RequeueAfter: time.Second * 10}, nil
+}
+
+func (r *EvalJobReconciler) getPod(ctx context.Context, job *lmevalservicev1beta1.EvalJob) (*corev1.Pod, error) {
+	var pod = corev1.Pod{}
+	if err := r.Get(ctx, types.NamespacedName{Namespace: job.Namespace, Name: job.Name}, &pod); err != nil {
```

Comment:
This looks like it's using the name of the `job` as the name of the `pod`? Is that always going to be the case? Seems like we might want to do some kind of hash suffix to correlate with the kind of naming used for other parent/child resource types like `Job` or `ReplicaSet`. This would be a nice shorthand for users to identify the pod as "machine created" vs "human created."

Reply:
since the pod name is stored in the status subresource. we definitely can have our own naming logic. apology for lazy on this in the first draft. :-p. I put the `OwnerReference` in the pod, so even from the pod, we know which job it belongs to.